src/getVulnerabilityByCve.js

```diff
@@ -0,0 +1,59 @@
+const requestWithDefaults = require('./requestWithDefaults');
+const { getTokenFromCache } = require('./tokenCache');
+const { getLogger } = require('./logger');
+const { RequestError } = require('./responses');
+
+/**
+ * Fetches vulnerability details from CrowdStrike Spotlight API for a given CVE ID.
+ * @param {string} cveId - The CVE ID to filter on (e.g., 'CVE-2023-12345').
+ * @param {object} options - Integration options (auth, etc).
+ * @returns {Promise<object>} - Structured API response data
+ */
+async function getVulnerabilityByCve(cveId, options) {
+  const Logger = getLogger();
+  // Set token
+  const token = getTokenFromCache(options);
```

Comment:
You don't want to access the token cache directly here.  This just pulls a token from the cache but if there is no token yet it this call will never work (try restarting your integration and only looking up a CVE, the integration won't be able to authenticate and will throw an error).

I'm guesing this worked for you because you searched an IP or hostname first which would cause a token to get cached.

Instead you want to use the `authenticatedRequest` method which will automatically handle getting a token for you and applying it to the request (if the token is not in the cache it will generate a new one).

You use it like this:

```
const response = await authenticatedRequest(requestOptions, options);
```

You can see in example in the other methods that fetch data.